lib/enrichment/src/find_enrichment_table_records.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{
+    vrl_util::{self, add_index},
+    Condition, IndexHandle, TableRegistry, TableSearch,
+};
+use std::collections::BTreeMap;
+use vrl_core::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct FindEnrichmentTableRecords;
+impl Function for FindEnrichmentTableRecords {
```

Comment:
Should we add this to the benchmarks as well? I realize they might not support enrichment tables yet in the `bench_function` macro so we may need to add that first.

Reply:
These functions are largely just a wrapper over the enrichment_table/file for which there are a number of benchmarks. But you are right, we should add them in case there are edge cases around the use of this function. I'll add it to the list of enrichment table issues.